inst/include/dust/interface_helpers.hpp

```diff
@@ -363,6 +364,33 @@ std::vector<size_t> check_step_snapshot(cpp11::sexp r_step_snapshot,
   return step_snapshot;
 }
 
+
+inline dust::cuda::device_config device_config(cpp11::sexp r_device_config) {
+  size_t run_block_size = 128;
+
+  cpp11::sexp r_device_id = r_device_config;
+  if (TYPEOF(r_device_config) == VECSXP) {
+    cpp11::list r_device_config_l = cpp11::as_cpp<cpp11::list>(r_device_config);
+    r_device_id = r_device_config_l["device_id"]; // could error if missing?
+    cpp11::sexp r_run_block_size = r_device_config_l["run_block_size"];
```

Comment:
Do we check this is a multiple of the warp size somewhere? (not an error, but definitely not a good idea if not % 32)

Reply:
I'm fine with making this an error, will do that now